src/geometry.rs

```diff
@@ -0,0 +1,188 @@
+//!
+//! Geometry helper functionality.
+use crate::{Vec3, Vec3u};
+
+/// A plane which can be intersected by a ray.
+#[derive(Debug, Copy, Clone)]
+pub struct Plane {
+    /// f32he plane described as x,y,z normal
+    normal: Vec3,
+    /// dot product of the point and normal, representing the plane position
+    bias: f32,
+}
+impl Plane {
+    /// Create a new `Plane`.
+    pub fn new(normal: Vec3, bias: f32) -> Self {
+        Plane { normal, bias }
+    }
+
+    /// Create a new `Plane` from a point normal representation
+    pub fn from_point_normal(point: &Vec3, normal: &Vec3) -> Self {
+        let normalized = normal.normalized();
+        Self {
+            normal: Vec3::new(normalized.x, normalized.y, normalized.z),
+            bias: point.dot(normalized),
+        }
+    }
+
+    /// Create a new `Plane` from a point normal representation
+    pub fn from_point_vectors(point: &Vec3, v1: &Vec3, v2: &Vec3) -> Self {
+        Self::from_point_normal(point, &v1.cross(*v2))
+    }
+
+    /// Create a `Plane` which is facing along the X-Axis at the provided coordinate.
+    pub fn with_x(x: f32) -> Self {
+        Self::from_point_normal(&Vec3::new(x, 0.0, 0.0), &Vec3::new(1.0, 0.0, 0.0))
+    }
+
+    /// Create a `Plane` which is facing along the Y-Axis at the provided coordinate.
+    pub fn with_y(y: f32) -> Self {
+        Self::from_point_normal(&Vec3::new(0.0, y, 0.0), &Vec3::new(0.0, 1.0, 0.0))
+    }
+
+    /// Create a `Plane` which is facing along the Z-Axis at the provided coordinate.
+    pub fn with_z(z: f32) -> Self {
+        Self::from_point_normal(&Vec3::new(0.0, 0.0, z), &Vec3::new(0.0, 0.0, 1.0))
+    }
+
+    /// f32his `Plane` normal
+    pub fn normal(&self) -> &Vec3 {
+        &self.normal
+    }
+
+    /// Normalized representation of this `Plane`
+    pub fn normalize(&self) -> Self {
+        let distance = self.normal.mag();
+        Self {
+            normal: self.normal / distance,
+            bias: self.bias / distance,
+        }
+    }
+
+    /// Returns the dot product of this `Plane` and a provided `Vec3`
```

Comment:
Doc comment for this and `dot` and `dot_plane` should tell a little more about what it does/what the difference between the two methods is.